### ISSUE TITLE
markdown-cycle shouldn't enter a new row unless at the end of the table

### ISSUE THREAD

**Issue by divansantana** [jrblevin/markdown-mode#522]
## Expected Behavior

Pressing <tab> which calls markdown-cycle command when you are at the
last column in a row (while in a table), should not create a new row,
unless you are at the end of the table, then it should create a new
row.  This is the expected behavior and this is how orgmode works.

If one wants to enter a new row, one should call C-o which should
create this new row.  Also in line with orgmode behavior.

It is confusing for an emacs user, because one expects the orgmode and
markdown mode tables to behave similarly.

## Actual behavior

When you are in a table, say at near the top of a table, and at the
end of the row, pressing <tab> inserts a new row, and moves point
there.  Instead I would expect the table to be realigned and my point
to be moved to the first cell on the next row.

## Software Versions

```
markdown-mode, version 2.5-dev
GNU Emacs 26.3 (build 1, x86_64-unknown-linux-gnu, GTK+ Version 3.24.20)
```

**Comment by syohex** [comment]
Sorry I introduced this bug at #490. I'll tackle this problem this weekend.

**Comment by syohex** [comment]
Thanks for reporting issue. I have merged #527. Please check latest version.

**Comment by divansantana** [comment]
@syohex great thanks!
